### ISSUE TITLE
[Bug]: Papillon utilise le mauvais fuseau horaire

### ISSUE THREAD

**Issue by hermitoff** [PapillonApp/papillon-v5#170]
### Description du bug

Bonjour,
Je suis dans le fuseau horaire UTC+11, et j’ai remarqué que ce n’est qu’à partir de 11:00 heure AM que Papillon se mets à utiliser les donnés Pronote d’aujourd’hui, par exemple avant 11h il ne m’affiche pas les cours à venir mais après 11h, il m’indique correctement l’heure et la durée avant le prochain cours. En faite ce n’est qu’à partir de 11h qu’il utilise les donnés du bon jour.

![IMG_5915](https://github.com/PapillonApp/Papillon/assets/82811865/a2cffb00-4d76-46dd-a5c0-15dbbec22bb6)

Ici le cours à suivre s’affiche correctement ce qui n’étais pas le cas il y’a 1h ou l’application ne détectais aucun cours à venir.

### Etapes à reproduire

1. Être dans un autre fuseau horaire que celui de l’API (Paris)
2. Ouvrir l’application et s’apercevoir que l’application essaye d’utiliser les donnés d’hier

### Comportement attendu

L’application devrais utiliser le fuseau horaire du client non de l’API, c’est à dire simplement afficher les cours à venir 

### Appareil

iPhone 8 Plus

### Système d`exploitation

IOS 16

### Version

5.8.3-ios

### Environnement

Application native iOS

### Service scolaire

🦋 Pronote

### Contexte supplémentaire

Ce bug ne touche pas seulement la page d’accueil mais également par exemple l’emploi du temps qui affiche le jour d’avant, il y’a un décalage avant 11h du coup même en glissant sur le côté il y’a toujours un décalage entre l’emploi du temps et la date affichée en haut à droite

**Comment by sivelswhy** [comment]
Effectivement, il faudrait fix ça. De plus, à la rentrée j'aurais un fuseau horaire différent que la france métro. donc un peu chiant

**Comment by moeenio** [comment]
du discord :
> Vexicted — 10/07/2023 13:36
j'ai oublié comment elle renvoie les dates
je crois que c'est genre un truc un peu chéper
t'as juste à trouver la timezone du serveur (qui est plutôt simple car dans la première réponse API de Pronote t'as la dateServeur qui correspond littéralement à l'heure où t'as fais la requête dans leur système donc bah tu peux facilement comparer et déterminer la timezone)
> et après tu modifies toutes les dates plus tard manuellement avant de les passer dans un new Date() ou whatever lib que vous utilisez

> Vincent - ecnivtwelve — 10/07/2023 13:42
j’obtiens ça comment avec pronotepy vu que j’ai pas la réponse de cette requête

> Vexicted — 10/07/2023 13:42
ah bah c'est à toi de trifouiller dans pronotepy
tajoutes ça jsp comment dans ton api 😭
mais en tout cas yes c'est possible
après reste à voir si y a pas un autre endroit où tu peux choper la date serveur
mais voilà c'est [là](https://github.com/catto-labs/pawnote/blob/44b234700f0e02a9110f18dcc9836d8d01758601/packages/api/src/handlers/login/informations/types.ts#L13) si jamais
(LoginInformations c'est mon équivalent de FonctionParametres)

**Comment by hermitoff** [comment]
Je me suis aperçu en hébergeant ma propre instance d'API a seulement 1h de différence de moi que cela n'influe pas sur le bug du temps, c'est donc un bug client qui demande la mauvaise heure je pense, je n'ai pas pu essayer sur la v6 la présence de ce problème.
